### DISCUSSION TITLE
Importing previously acquired DICOMs by sequence into NIDB

### DISCUSSION THREAD

**Discussion by jcipser** [gbook/nidb#109] [General]
Hello Greg,

I am excited about getting NIDB installed and running on a server at my host institution (University of Cape Town). I have been testing a pilot version installed locally on a laptop. 

Up to this point I have been focusing on importing previously acquired DICOM files, and was wondering what the most efficient method might be to do this. 

Within each subject folder, my DICOMs  for each sequence (eg. T1w, rs-fMRI) are located within different subfolders. What I have been doing is copying DICOM files for a particular subject and specific sequence into the dicomincoming folder, and once these files have been detected and imported into NIDB, repeating this procedure for all other sequences. I have taken this approach as it does not appear as if NIDB is able to recursively process DICOM files within subfolders in the  dicomincomging folder. Is this correct?

Best,
Jonathan

**Comment by gbook** [comment]
Hi Jonathan,

That's correct, the import module does not read sub-directories when you dump dicoms into the import directory.

There are 2 methods to batch import DICOM data into NiDB:
1) Global method - used by dicom receiver. All files go into the same directory and are archived automatically
2) Individual import - import a single zip file or directory (can contain sub-directories). Parses subject/study/series before archiving

The global method is easiest to import a large batch of existing DICOMs. Dump your dicom files in `/nidb/data/dicomincoming` directory, which is what you were doing. **If the files are all uniquely named**, you can run the following command to copy files into the import directory. Run the command from the directory containing the dicom files/directories.

`find . -name '*.dcm' -exec cp {} /nidb/data/dicomincoming \;`

You can view some details of the import process under Admin &rarr; Modules &rarr; import &rarr; View logs

To import **individual batches of data**, go to Data &rarr; Import Imaging and follow the steps to create a new import. You will need to select the series to import after the parsing is done.

**Reply by jcipser** [reply]
Hello Greg,

Thank you for the clarification (and apologies for the delayed reply - I have been on leave and more-or-less  out-of-touch for the last couple of weeks). I can confirm that I was able to import all of the DICOM sequences for a test subject using the directions you provided for the global method. The trick seems to be to import the DICOM files for all of the sequences acquired for a specific subject at a scan visit in one go. Previously I tried importing the DICOMS for all subjects for a project by sequence (eg. first importing files for a T1w sequence, then a rs-fMRI sequence, etc.), which made matching subsequent sequences to particular subjects more difficult than it needed to be, it seems. 

Thanks again for the help,
Jonathan

**Reply by gbook** [reply]
NiDB should be able to import DICOMs from a group of subjects without issue. It can handle diverse data should archive them correctly, but it depends on the DICOM tags. It organizes data by `PatientID` tag for the subject, then by the `StudyDate` & `StudyTime` tags for the study, then by `SeriesNumber` tag for the series. I'm wondering if some of the DICOM tags are missing, or identical? Can you check the headers?

For example, if the `PatientID` tag is the same (or blank) for every subject, then all of those tags are assumed to be part of the same subject. If the unique ID is stored in the `PatientName` tag for example, instead of the `PatientID` tag, this can causes issues.

The same applies to the `StudyDate` and `StudyTime` tags. If they are blank, or partially anonymized, NiDB will assume they are all part of the same imaging study. Or if `SeriesNumber` tag is missing or blank; same result for the series.

There is some information about the parsing/archiving process in the import log files. This can give an idea what NiDB is encountering and what it's trying to do with it. That's under **Admin** --> **Modules** --> **import** --> **View Logs**.

You can also try importing through the website, using the **Data** --> **Import Imaging** tool. Try importing 2 or 3 subjects. It will parse the data before attempting to archive it, and it will show you exactly what subject/study/series were found in the data. If all of the series end up being associated with the same subject or study, then we know there's a blank or non-unique DICOM tag.

To check tags: On Windows, there's a useful tool called MicroDICOM, which allows viewing of DICOM tags. On Linux, there's a command line tool called exiftool which will display all the DICOM tags.
It is installed by default when NiDB is installed. Run it from the terminal with `exiftool thedicomfile.dcm`

**Reply by jcipser** [reply]
Thank you for the tips. I recently also came across Dicombrowser (https://wiki.xnat.org/xnat-tools/dicombrowser) which seem to be useful for modifying tag values in the DICOM headers, and handily supports renaming values across image sequences for the same session.
